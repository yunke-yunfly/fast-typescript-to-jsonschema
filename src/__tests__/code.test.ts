import genTypeSchema from '../index';

const code = `
import AAA from './aaa'
export type Generic_10 = Generic_type_1<string>
export interface Generic_type_1<T> {
    value1: string;
    value2: T;
}
interface BBB {
    some: string;
}
export enum Role {
  Maintainer = 'Maintainer',
  Developer = 'Developer',
}
type Page = 'home' | 'about' | 'contact' | string;
interface PageInfo {
  title: string;
  some: BBB
}
`
genTypeSchema.genJsonDataFromCode(code);

const getSchema = (type: string) => {
  const schema = genTypeSchema.getJsonSchema(type);
  return schema;
};

test('枚举', () => {
  expect(getSchema('Role')).toMatchSnapshot();
});

test('Type类型', () => {
  expect(getSchema('Page')).toMatchSnapshot();
});

test('简单接口', () => {
  expect(getSchema('BBB')).toMatchSnapshot();
});

test('嵌套接口', () => {
  expect(getSchema('PageInfo')).toMatchSnapshot();
});

test('泛型', () => {
  expect(getSchema('Generic_type_1')).toMatchSnapshot();
});

test('嵌套泛型', () => {
  expect(getSchema('Generic_10')).toMatchSnapshot();
});

