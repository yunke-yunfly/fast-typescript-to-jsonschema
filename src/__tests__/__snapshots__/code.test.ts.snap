// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type类型 1`] = `
Object {
  "anyOf": Array [
    Object {
      "type": "string",
    },
    Object {
      "enum": Object {
        "enum": Array [
          "home",
          "about",
          "contact",
        ],
      },
      "type": "string",
    },
  ],
}
`;

exports[`嵌套接口 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "BBB": Object {
      "additionalProperties": false,
      "properties": Object {
        "some": Object {
          "type": "string",
        },
      },
      "required": Array [
        "some",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "some": Object {
      "$ref": "#/definitions/BBB",
    },
    "title": Object {
      "type": "string",
    },
  },
  "required": Array [
    "title",
    "some",
  ],
  "type": "object",
}
`;

exports[`嵌套泛型 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {},
  "properties": Object {
    "value1": Object {
      "type": "string",
    },
    "value2": Object {
      "type": "string",
    },
  },
  "required": Array [
    "value1",
    "value2",
  ],
  "type": "object",
}
`;

exports[`枚举 1`] = `
Object {
  "enum": Array [
    "Maintainer",
    "Developer",
  ],
  "type": "string",
}
`;

exports[`泛型 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "T": Object {},
  },
  "properties": Object {
    "value1": Object {
      "type": "string",
    },
    "value2": Object {
      "$ref": "#/definitions/T",
    },
  },
  "required": Array [
    "value1",
    "value2",
  ],
  "type": "object",
}
`;

exports[`简单接口 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "some": Object {
      "type": "string",
    },
  },
  "required": Array [
    "some",
  ],
  "type": "object",
}
`;
